src/Components/ViewCart/ViewCart.component.js

```diff
@@ -0,0 +1,183 @@
+
+import { SEND_RECENT_ITEM_SELECTED_API, USER_TOKEN } from "../../config";
+
+import {
+  toggleAllSelectCheckBox,
+  deleteSelectedItems,
+  selectedItemsTotalPrice,
+  checkStatusAllSelectCheckBox,
+  getSelectedItemsAmount,
+} from "../../redux/cart/cart.actions";
+
+import "./ViewCart.styles.scss";
+
+class ViewCart extends Component {
+  componentDidMount() {
+    const { checkStatusAllSelectCheckBox, getSelectedItemsAmount } = this.props;
+    checkStatusAllSelectCheckBox();
+    getSelectedItemsAmount();
+  }
+
+  AllSelectCheckboxClick = () => {
+    const {
+      toggleAllSelectCheckBox,
+      selectedItemsTotalPrice,
+      getSelectedItemsAmount,
+    } = this.props;
+    toggleAllSelectCheckBox();
+    selectedItemsTotalPrice();
+    getSelectedItemsAmount();
+    fetch(SEND_RECENT_ITEM_SELECTED_API, {
+      method: "PUT",
+      headers: {
+        "content-type": "application/json",
+        Authorization: USER_TOKEN,
+      },
+      body: JSON.stringify({
+        selected: "all",
+      }),
+    })
+      .then((response) => response.json())
+      .catch((error) => console.log(error));
+  };
+
+  selectedItemDeleteCheckboxCLick = () => {
+    const {
+      deleteSelectedItems,
+      selectedItemsTotalPrice,
+      checkStatusAllSelectCheckBox,
+      getSelectedItemsAmount,
+    } = this.props;
+    deleteSelectedItems();
+    selectedItemsTotalPrice();
+    checkStatusAllSelectCheckBox();
+    getSelectedItemsAmount();
+    fetch(SEND_RECENT_ITEM_SELECTED_API, {
+      method: "DELETE",
+      headers: {
+        "content-type": "application/json",
+        Authorization: USER_TOKEN,
+      },
+      body: JSON.stringify({
+        delete: "selected",
+      }),
+    })
+      .then((response) => response.json())
+      .catch((error) => console.log(error));
+  };
+
+  soldOutItemDeleteButtonClick = () => {
+    const {
+      filterOutSoldoutItems,
+      selectedItemsTotalPrice,
+      checkStatusAllSelectCheckBox,
+      getSelectedItemsAmount,
+    } = this.props;
+    filterOutSoldoutItems();
+    selectedItemsTotalPrice();
+    checkStatusAllSelectCheckBox();
+    getSelectedItemsAmount();
+    fetch(SEND_RECENT_ITEM_SELECTED_API, {
+      method: "DELETE",
+      headers: {
+        "content-type": "application/json",
+        Authorization: USER_TOKEN,
+      },
+      body: JSON.stringify({
+        delete: "soldout",
+      }),
+    })
+      .then((response) => response.json())
+      .catch((error) => console.log(error));
+  };
+
+  render() {
+    const { cartItems, allSelect, seletectedItemsAmount } = this.props;
+    return (
+      <div className="ViewCart">
+        <div className="selected-items">
+          <div className="selected-items-header">
+            <div className="selected-items-header-select">
+              <label>
+                <input
+                  type="checkbox"
+                  onChange={this.AllSelectCheckboxClick}
+                  checked={allSelect ? true : false}
```

Comment:
```suggestion
                  checked={!!allSelect}
```
- 결국 true/false를 사용하게 되는데 allSelect라는 값이 애초에 boolean 값인듯 하네요.
- 부정연산자 두번 붙이면 어떤 값이든 boolean을 갖게 됩니다.

Reply:
수정했습니다! 😄